cas/service.c

```diff
@@ -1326,6 +1373,35 @@ static int cas_fom_tick(struct m0_fom *fom0)
 			m0_ctg_op_init(&fom->cf_ctg_op, fom0,
 				       cas_op(fom0)->cg_flags);
 		}
+
+		/* If cas request has a txr payload, this a dtm0 operation.
+		 * We need to calculate credits for creating a dtm0 log record.
+		 *
+		 */
+		if (!m0_dtm0_tx_desc_is_none(&cas_op(fom0)->cg_txd)) {
+			struct m0_be_tx_credit dtm0logrec_cred;
+			struct m0_buf	       buf = {};
+			int	    rc;
+
+			/* TDB: while calculating the credits we need only the
+			   size of the payload. Check if m0_be_dtm0_log_credit
+			   needs to be updated to accept a size instead of the
+			   actual payload. */
+
+			rc = m0_xcode_obj_enc_to_buf(&M0_XCODE_OBJ(m0_cas_op_xc,
+						      cas_op(fom0)), &buf.b_addr, &buf.b_nob);
+			M0_ASSERT(rc == 0);
```

Comment:
I think this error can and should be handled instead of panic.

Reply:
Handled